lib/src/rules/avoid_web_libraries_in_flutter.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/file_system/file_system.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/ast.dart';
+import 'package:yaml/yaml.dart';
+
+const _desc = r'Avoid using web-only libraries outside Flutter web projects.';
+
+const _details = r'''Avoid using web libraries, `dart:html`, `dart:js` and 
+`dart:js_util` in non-web Flutter projects.  These libraries are not supported
+outside a web context and functionality that depends on them will fail at
+runtime.
+
+Web library access is allowed in:
+
+* projects meant to run on the web (e.g., have a `web/` directory)
+* plugin packages that declare `web` as a supported context
+
+otherwise, imports of `dart:html`, `dart:js` and  `dart:js_util` are flagged.
+''';
+
+const _webLibs = [
```

Comment:
Should this be a set?

Reply:
Not clear whether the time to perform a hashed lookup would be faster than a linear search with only three items. It might even be faster to write something like
```
int uriLength = uri.length;
if ((uriLength == 9 && uri == 'dart:html') ||
    (uriLength == 7 && uri == 'dart:js') ||
    (uriLength == 12 && uri == 'dart:js_util')) {
  // ...
}

Reply:
Good deal.  Will update!